tests/testbed_validation_rules.py

```diff
@@ -0,0 +1,85 @@
+from abc import ABC, abstractmethod
+import re
+import logging
+from pyats.topology.testbed import Testbed
+
+
+_log = logging.getLogger(__name__)
+
+
+class Rule(ABC):
+
+    ip_pattern = r"^\d{3}\.\d{3}\.\d{1,3}\.\d{1,3}$"
+
+    @abstractmethod
+    def is_passed(self, testbed: Testbed) -> bool:
+        """Provides a result of rule execution"""
+        pass
+
+    @abstractmethod
+    def __str__(self):
+        """Returns rule name"""
+        pass
+
+
+class CheckUserCredentials(Rule):
+    def is_passed(self, testbed: Testbed) -> bool:
+        for device in testbed:
+            device.connect()
+            assert device.connected
+        return True
+
+    def __str__(self):
+        return "Check user credentials"
+
+
+class CheckDevicesIp(Rule):
+    def is_passed(self, testbed: Testbed) -> bool:
+        for device in testbed:
+            device_ip = str(device.connections.main.ip)
+            assert re.fullmatch(self.ip_pattern, device_ip)
+        return True
+
+    def __str__(self):
+        return "Check devices ip format"
+
+
+class CheckSeleniumGridUrl(Rule):
+
+    url_pattern = r"^https?:\/\/.+:\d+$"
+
+    def is_passed(self, testbed: Testbed) -> bool:
+        grid_url = testbed.custom["selenium-grid"]
+        assert re.fullmatch(self.url_pattern, grid_url)
```

Comment:
@SergiyBeno I think 
```suggestion
        return re.fullmatch(self.url_pattern, grid_url)
```
is more suitable.

And please check it for other rules also.

Reply:
Yes, indeed - I'll replace.